UI/window-projector.cpp

```diff
@@ -17,11 +17,14 @@ static bool updatingMultiview = false, mouseSwitching, transitionOnDoubleClick;
 
 OBSProjector::OBSProjector(QWidget *widget, obs_source_t *source_, int monitor,
 			   ProjectorType type_)
-	: OBSQTDisplay(widget, Qt::Window),
-	  source(source_),
-	  removedSignal(obs_source_get_signal_handler(source), "remove",
-			OBSSourceRemoved, this)
+	: OBSQTDisplay(widget, Qt::Window), weakSource(OBSGetWeakRef(source_))
 {
+	OBSSource source = GetSource();
+
+	if (source)
+		removedSignal.Connect(obs_source_get_signal_handler(source),
```

Comment:
Since we're using `destroy` instead of `remove`, we should probably change the name of this. Other than that, this looks good provided we don't get memory leaks.

Reply:
Updated the name of the function and the signal.